source/crud/crud.rst

```diff
@@ -644,7 +644,7 @@ Basic
      * @see https://docs.mongodb.com/manual/reference/command/update/
      * @throws WriteException
      */
-    replaceOne(filter: Document, replacement: Document, options: Optional<UpdateOptions>): UpdateResult;
+    replaceOne(filter: Document, replacement: Document, options: Optional<ReplaceOptions>): UpdateResult;
```

Comment:
@rstam: In [SPEC-939](https://jira.mongodb.org/browse/SPEC-939) you mentioned:

> This does have backward compatibility implications that need to be addressed.

The spec has permissible language about the structure of options in Operation Parameters › Deviations, so BC concerns only exist for drivers that actually use classes for these. This seems like something that could be handled on a case-by-case basis -- I don't think we need a FAQ entry about it in the spec.

Reply:
In .NET I imagine we will need to add a new overload of the ReplaceOne method that takes a ReplaceOptions instead of an UpdateOptions.

The existing ReplaceOne method that takes an UpdateOptions would be deprecated, and would construct a new ReplaceOptions from the UpdateOptions and delegate to the new method.

It might be worth at least mentioning that backward compatibility is an issue, even if you don't intend to provide guidance on how to achieve it.

Reply:
Added a note to the changelog entry.